### PR TITLE
TH-174: Aura > Cart Page > Format product price by the Dotshop object's locale code

### PR DESCRIPTION
**Pull request by eihabkhan1**
## JIRA Ticket

Ticket: [TH-174](https://youcanshop.atlassian.net/browse/TH-174).

## Prerequisites

* [ ] Have Aura active

### Multi Currency **OFF**
* [ ] From the multi currency settings in seller area, turn off multi currency and enable precision
* [ ] Go to your store's general settings `/admin/settings/general#details`
* [ ] Change the store language to `English - United States`
* [ ] Change the store currency to `Moroccan Dirhams`
* [ ] Change the currency symbol to `MAD`
* [ ] Go to your product page
* [ ] Add it to the cart
* [ ] Add another product to your cart
* [ ] From the cart drawer, click on "Checkout" to go to the cart page
* [ ] The prices below should be correctly formatted:

Screenshot 2024-10-21 at 3 34 43 PM

* [ ] Go back to your store, change the language to `French - France`
* [ ] Go back to storefront / cart page, hit refresh
* [ ] The price should be formatted accordingly and the currency symbol should be on the right
* [ ] Go back to your store, this time change the currency symbol to a random currency symbol (ex: `Wiwi`)
* [ ] Go back to storefront, hit refresh, and open the cart
* [ ] The price should be formatted accordingly and the currency symbol should still be on the right

### Multi Currency **ON**
* [ ] From the multi currency settings in seller area, turn on multi currency and enable precision
* [ ] Turn on your VPN
* [ ] Go to your product's page in storefront
* [ ] Add it to the cart
* [ ] Make sure that:
    * [ ] The price for each item in the cart is formatted based on your VPN's country + your browser's language
    * [ ] The price is converted to your VPN's country currency
    * [ ] The currency symbol displayed is the VPN's country currency symbol
    * [ ] The subtotal is also formatted accordingly
* [ ] Go to your browser's language settings, If you're using Chrome just click `⌘ + ,`
* [ ] Search for language
Screenshot 2024-10-08 at 8 43 12 AM

* [ ] Click the three dots next to any language and move it to the top to make it your language of choice
* [ ] Go back to your storefront, hit refresh and open your cart
* [ ] The price value(s) and subtotal should be formatted based on your VPN + your browser's language

> [!Note]
> Note that if your IP address while using VPN is in UK, and you have Arabic as the browser language, then you might see a flash where the price is displayed in indic numerals at first then Arabic numbers `١٢٣.٩٩ ' -> `123.99`. This is likely an issue with how PHP handles ar-GB vs how javaScript handles it. Will tackle this issue in a separate PR.

## Note

Leave empty when you have nothing to say.

[TH-174]: https://youcanshop.atlassian.net/browse/TH-174?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by eihabkhan1** [comment]
cart.js needs to be refactored: https://youtu.be/mWMFTfaJaWM

**Comment by ibrilBadreddine** [comment]
> cart.js needs to be refactored: https://youtu.be/mWMFTfaJaWM

https://www.youtube.com/watch?v=PUhncAezGUQ

**Review by ibrilBadreddine — COMMENTED** [review]
> [!WARNING]
> I tried QA with the `add to cart` feature on, but clicking any product card (home or collections) triggered this error: 

`add-to-cart.js` -> `directAddToCart` function
```
Uncaught (in promise) ReferenceError: trackVariantQuantityOnCart is not defined
```

File: themes/aura/sections/main-cart.liquid [hunk 1]
```diff
@@ -6,7 +6,7 @@
   <div class="header-block">
     <div class="header-bar">
       <h2>{{ 'cart.title' | t }}</h2>
-      <p class='mt-1'>
+      <p class='mt-1' id="cart-items-count">
```
[the remark this answers is not in the document]
Comment (bj-anas) [reply themes/aura/sections/main-cart.liquid:9]: it is used in js also https://github.com/youcan-shop/youcan-themes/blob/ddd79855f3439f73a4c1815bda64170455d39106/themes/harmony/assets/cart.js#L129

File: themes/aura/assets/main.js [hunk 2]
```diff
@@ -336,8 +336,9 @@ function formatCurrency(amount, currencySymbol, locale = 'en-US') {
   const usePrecision = shouldUsePrecision(amount);
   const formatter = new Intl.NumberFormat(locale, {
     style: 'decimal',
+    roundingMode: 'floor',
     maximumFractionDigits: usePrecision ? 2 : 0,
-    roundingMode: 'floor'
+    minimumFractionDigits: usePrecision ? 2 : 0,
```
[the remark this answers is not in the document]
Comment (bj-anas) [reply themes/aura/assets/main.js:341]: max & min number is 2 for a floating number

**Comment by bj-anas** [comment]
> Warning
> 
> I tried QA with the `add to cart` feature on, but clicking any product card (home or collections) triggered this error:
> 
> `add-to-cart.js` -> `directAddToCart` function
> 
> ```
> Uncaught (in promise) ReferenceError: trackVariantQuantityOnCart is not defined
> ```

You have some problems with syncing files while you running the dev mode in your local machine

**Review by ibrilBadreddine — APPROVED** [review]

**Pull request merged**